File: src/main/go/com/lmmmowi/leetcode/go/p136/Solution.go
// 136.只出现一次的数字[https://leetcode-cn.com/problems/single-number/]
package main

import "fmt"

func main() {
	x := singleNumber([]int{2, 2, 1})
	fmt.Print(x)
}

func singleNumber(nums []int) int {
	result := 0
	for i := 0; i < len(nums); i++ {
		result ^= nums[i]
	}
	return result
}

File: src/main/java/com/lmmmowi/leetcode/p136/Solution.java
package com.lmmmowi.leetcode.p136;

/**
 * @Author: lmmmowi
 * @Date: 2021/4/6
 * @Description: 136. 只出现一次的数字[https://leetcode-cn.com/problems/single-number/]
 */
public class Solution {

    public int singleNumber(int[] nums) {
        int r = 0;
        for (int i = 0; i < nums.length; i++) {
            r = r ^ nums[i];
        }
        return r;
    }
}
